stable/oauth2-proxy/templates/deployment.yaml

```diff
@@ -65,6 +65,7 @@ spec:
         {{- if .Values.htpasswdFile.enabled }}
           - --htpasswd-file=/etc/oauth2_proxy/htpasswd/users.txt
         {{- end }}
+        {{- if not .Values.useEnvironmentNotSecrets }}
```

Comment:
I think that the name of this flag is confusing for 2 reasons:
1. there is a double negation
2. if set to `true` then it drops the environment variables section

Can you find another name?

Reply:
1) Thanks for your reasonable point. I have changed the name and reversed the meaning of the boolean.
2) Our use of this chart. Instead of k8s secrets we use a setup with vault from hashicorp and vault-secrets-webhook from banzaicloud. Secrets are referred to as vault paths. When a deployment that has the secret annotation is been deployed, the vault-secrets-webhook is activated. On behalf of the pod the vault-secrets-webhook exchanges the vault path with the secret stored in vault. Now the pod is ready to start.
Secrets in vault for a specific pod may only be read with this pods credentials.
With this setup secrets are not readable in a cluster backup and cluster admins are not able to read secrets.
When we disable the secrets in this chart we add the three variables in extraEnv. Where the value are vault paths.